text/0003-blockspace-regions.md

```diff
@@ -0,0 +1,252 @@
+```
+
+This API is intended to be wrapped by logic implementing interfaces such as RFC-5.
+
+Created regions for a core MUST NOT exceed a combined block rate of 1 at any block.
+
+### Changes to backing/availability
+
+Regions are used to modify the behavior of the parachain backing/availability pipeline. The first major change is that parachain candidates submitted to the relay-chain in the `ParasInherent` will be annotated with the `RegionId` that they are intended to occupy. Validators and collators do the work of figuring out which blocks are assigned to which regions, lifting the burden of granular scheduling off of the relay chain.
+
+The **maximum consumption** of a region at any block number `now` is given by:
+`maximum_consumption(now, region) = (min(now, end) - start) * rate`
+If `end` is `None`, it is treated as infinite.
+
+The **minimum consumption** of a region at any block number `now` is given by:
+`minimum_consumption(now, region) = maximum_consumption(now - SCHEDULING_LENIENCE, region)`
+
+The **effective consumption** of a region at any block number `now` is the given by:
+`max(region.consumption, minimum_consumption(now, region))`
+
+A submitted candidate which uses `p: PartsOf57600` of the core's resources for a parachain P at a block B is accepted if:
+  * There is no candidate pending availability for B
+  * `effective_consumption(B, region) + p <= maximum` if `maximum` is `Some`
+  * `effective_consumption(B, region) + p <= maximum_consumption(B, region)`
+
+How core resource limits are defined is left beyond the scope of this RFC - at the time of writing, all cores have the same resource limits, but this design allows cores to be specialized in their resource limits, with some cores allowing more data, some allowing more granularity or execution time, etc.
+
+If all of these conditions are met, along with other validity conditions for backed candidates beyond the scope of this RFC, then the candidate is pending availability and the region's consumption value is incremented to `effective_consumption(B, region) + p` If the candidate times out before becoming available, the consumption value is reduced by `p`.
+
+The scheduling lenience allows regions to fall behind their expected tickrate, limited to a small maximum level of debt. This prevents accumulated core debt from being accumulated indefinitely and spent when convenient. Smoothing system load over short time horizons is desirable, but over infinite time horizons becomes dangerous.
+
+This RFC introduces a new `BURST_CORES` parameter into the `HostConfiguration` which relay-chain governance uses to manage the parameters of the parachains protocol. BURST_CORES are inspired by technologies such as hyperthreading, to emulate multiple logical cores on a single physical core as resources permit. BURST_CORES allow parachains to make up for missed scheduling opportunities, which is important to effectively decouple parachain growth from backing on the relay chain.
```

Comment:
> while it would be harder to handle untrusted collators for abusing burst cores.

They are limited by `BURST_CORES_PER_CORE`, as well

Reply:
One of the key goals of the RFC is to reduce friction between parachains that are scheduled on the same core. At the moment, there is no friction because each parachain has a single dedicated core.

Reducing or eliminating friction should be a key goal of the system, as the system's value is driven by the quantity of processes using it and the efficient allocation of resources across those processes.

Reply:
One of the main reasons for this friction is that candidates pin to a specific relay-parent, and that the set of allowed relay parents expires relatively quickly.

> But even in this case the para could fix this by itself by ordering an on-demand core for making up for the missed opportunity caused by a malicious backing group

This would be a terrible experience for users - it reduces the value proposition of acquiring a core in the first place.